[executorch] Make trymerge force merges actually work with executorch

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #121921
* __->__ #121920

This PR addresses an issue with the trymerge function for executorch, which currently uses Facebook CLA instead of Easy CLA. This bug has been patched in #121921. However, the patch is potentially controversial, and we still want to verify Facebook CLA if it exists. Therefore, this PR includes Facebook CLA in our set of mandatory checks.

Additionally, this PR removes Facebook CLA from one of the mocks. This change is necessary because the specific PR used for testing fails due to the presence of Facebook CLA in the mock.

## Testing:
We run `find_matching_merge_rule(pr = GitHubPR("pytorch", "executorch", 2326), skip_mandatory_checks=True, skip_internal_checks=True)` to check if things work

https://pastebin.com/HHSFp2Gw

PaliC committed:
[executorch] Make trymerge force merges actually work with executorch

[ghstack-poisoned]

seemethere commented:
Would it make sense to eventually break this out to test-infra?

PaliC committed:
Update on "[executorch] Make trymerge force merges actually work with executorch"




[ghstack-poisoned]

huydhn commented:
> Would it make sense to eventually break this out to test-infra?

I thought about this.  The process would be a bit more involved than just moving the code because internal tool imports `read_merge_rules` and `find_matching_merge_rule` from PyTorch location.  They need to be updated.  But yeah, having merge bot reusable by other repos is great

huydhn commented on .github/scripts/test_trymerge.py:
```diff
@@ -205,7 +205,6 @@ def mocked_read_merge_rules(repo: Any, org: str, project: str) -> List[MergeRule
             approved_by=["pytorch/metamates", "ngimel"],
             mandatory_checks_name=[
                 "Lint",
-                "Facebook CLA Check",
```
As this is a mock value, this is probably ok.  The test you see fails because it caches an old PR https://github.com/pytorch/pytorch/pull/79694 where `Facebook CLA Check` was still in used?  Another way is to choose a newer example I think

PaliC replied:
Yeah, but I'm unsure how to put the new pr in. Also a good example is kindof hard to find without making it yourself.

PaliC commented:
@pytorchbot merge

PaliC committed:
Update on "[executorch] Make trymerge force merges actually work with executorch"


This PR addresses an issue with the trymerge function for executorch, which currently uses Facebook CLA instead of Easy CLA. This bug has been patched in #121921. However, the patch is potentially controversial, and we still want to verify Facebook CLA if it exists. Therefore, this PR includes Facebook CLA in our set of mandatory checks.

Additionally, this PR removes Facebook CLA from one of the mocks. This change is necessary because the specific PR used for testing fails due to the presence of Facebook CLA in the mock.

## Testing:
We run `find_matching_merge_rule(pr = GitHubPR("pytorch", "executorch", 2326), skip_mandatory_checks=True, skip_internal_checks=True)` to check if things work

https://pastebin.com/HHSFp2Gw


[ghstack-poisoned]

PaliC commented:
@pytorchbot merge
